VSC dev container for OED

huss opened the pull request:
# Description

This is a basic container that works with the Visual Studio Code Remote - Containers.
The container has the following features:

- You can use Docker in it.
- It has the extensions gitlens, code-spell-checker, azuretools and vscode-pull-request-github.

You can still use it outside Visual Studio Code as was done previously.

This was tested by developers running MacOS X and Windows 10. It ran fine.

This also adds to the .gitignore to exclude the new nohup.out file and .DS_Store. The later is Mac specific but this avoids it getting into the repo.

## Type of change

- [ ] Note merging this changes the node modules
- [ ] Note merging this changes the database configuration.
- [x] This change requires a documentation update

## Checklist

- [x] I have followed the [OED pull request](https://openenergydashboard.github.io/developer/pr.html) ideas
- [x] I have removed text in ( ) from the issue request

## Limitations

- This is a work in progress where some items are left commented out from what VSC created in case we want them later.
- The debugger is not working.
- The test data scripts and not yet changed. All scripts are planned to run inside the container rather than in the local machine shell. These changes are coming as the new developer documentation is completed.

huss committed:
VSC dev container for OED

This is a basic container that works with the VSC Remote - Containers.
The container has the following features:
- You can use Docker in it.
- It has the extensions gitlens, code-spell-checker, azuretools and
vscode-pull-request-github.

Limitiations:
- This is a work in progress where some items are left commented out
from what VSC created in case we want them later.
- The debugger is not working.
- The test data scripts and not yet changed. All scripts are planned
to run inside the container rather than in the local machine shell.

huss committed:
add license info to files

jina2k reviewed (approved):
Tested on Windows, and runs properly. As mentioned the basic container works, but running npm run testData is limited as it requires another container to add the data.

huss committed:
update install for VSC container

- remove docker compose from test data install since now done inside
container
- add date to install output so easier to see each one

huss commented:
@jina2k has reviewed this PR and approved. I push a few changes to fix up how the test data is done with new VSC container and added an output line at start of install so easier to see each one. I have updated the developer web pages to use the new VSC container. I consider it ready to merge.
